src/App.tsx

```diff
@@ -1,61 +1,129 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+function getUserByName(userName: string): User | undefined {
+  return usersFromServer.find(user => user.name === userName);
+}
 
 export const App = () => {
+  const [visibleTodos, setTodos] = useState(todos);
+  const [author, setAuthor] = useState('Choose a user');
+  const [task, setTask] = useState('');
+  const [isActiveTask, setIsActiveTask] = useState(false);
+  const [isActiveAuthor, setIsActiveAuthor] = useState(false);
```

Comment:
You don't really need these variables. You just can check if the title length is greater than 0 and you have the correct user ID